cpu/nrf5x_common/periph/uart.c

```diff
@@ -31,39 +33,68 @@
 #include "periph/gpio.h"
 
 #ifdef CPU_MODEL_NRF52840XXAA
-#define PSEL_RXD         NRF_UART0->PSEL.RXD
-#define PSEL_TXD         NRF_UART0->PSEL.TXD
-#define PSEL_RTS         NRF_UART0->PSEL.RTS
-#define PSEL_CTS         NRF_UART0->PSEL.CTS
+#define PSEL_RXD         dev(uart)->PSEL.RXD
+#define PSEL_TXD         dev(uart)->PSEL.TXD
+#define PSEL_RTS         dev(uart)->PSEL.RTS
+#define PSEL_CTS         dev(uart)->PSEL.CTS
+#define UART_IRQN        uart_config[uart].irqn
+#define UART_PIN_RX      uart_config[uart].rx_pin
+#define UART_PIN_TX      uart_config[uart].tx_pin
+#define UART_PIN_RTS     uart_config[uart].rts_pin
+#define UART_PIN_CTS     uart_config[uart].cts_pin
+#define UART_HWFLOWCTRL  (uart_config[uart].rts_pin != GPIO_UNDEF && \
+                          uart_config[uart].cts_pin != GPIO_UNDEF)
 #else
-#define PSEL_RXD         NRF_UART0->PSELRXD
-#define PSEL_TXD         NRF_UART0->PSELTXD
-#define PSEL_RTS         NRF_UART0->PSELRTS
-#define PSEL_CTS         NRF_UART0->PSELCTS
+#define PSEL_RXD         dev(uart)->PSELRXD
+#define PSEL_TXD         dev(uart)->PSELTXD
+#define PSEL_RTS         dev(uart)->PSELRTS
+#define PSEL_CTS         dev(uart)->PSELCTS
+#define UART_0_ISR       isr_uart0
+#ifndef UART_PIN_RTS
+#define UART_PIN_RTS     GPIO_UNDEF
+#endif
+#ifndef UART_PIN_CTS
+#define UART_PIN_CTS     GPIO_UNDEF
+#endif
+#ifndef UART_HWFLOWCTRL
+#define UART_HWFLOWCTRL  0
+#endif
 #endif
 
 /**
  * @brief Allocate memory for the interrupt context
  */
-static uart_isr_ctx_t uart_config;
+static uart_isr_ctx_t isr_ctx;
```

Comment:
just a single ISR context for (potentially) more than 1 configured UART interface? This seems broken to me. Probably this should be 
```c
static uart_isr_ctx_t isr_ctx[UART_NUMOF];
```
right?

Reply:
oops

Reply:
Fixed using preprocessor macros...